### DISCUSSION TITLE
Compound assignment for IEnumerable

### DISCUSSION THREAD

**Discussion by ivancavlek** [dotnet/csharplang#5833] [Language Ideas] [answered]
We have compound assignment for arithmetic operators:

`x op= y`

```
int i = 1;
i *= 2;
```

I believe it could be a good idea to have compound assignment for IEnumerables.
So for a situation like:

```
IEnumerable<string> strings = new List<string>();
strings.Add("test");
strings.Add("another test");
strings.Add("bla");
strings = strings.OrderBy(str => str.Length).ToList();
```

we would have something like:
`strings .= OrderBy(str => str.Length).ToList();`

It'd be a bit of shorthand notation where we wouldn't have to repeat the object name.

**Comment by svick** [comment, marked as answer]
This is a duplicate of https://github.com/dotnet/csharplang/discussions/500.
